exonum/src/node/consensus.rs

```diff
@@ -309,13 +337,19 @@ impl NodeHandler {
             if our_block_hash != block_hash {
```

Comment:
Sorry for off-topic, but I can't help but notice that the code above is immensely incomprehensible to me. (I think I've written about it ~a year ago, but the code hasn't disappeared somehow 🙃)

- Why does `take_unknown_propose_with_precommits` return a vector if the cycle returns after the first iteration? Is this an error with the cycle or an error with `take_unknown_propose_with_precommits`?
- Can there be multiple values in the returned vector? When?

Reply:
Indeed it seems that there can't be more than one element in the vector.
I renamed methods to `add_propose_confirmed_by_majority` / `take_confirmed_propose`, made them operate `Option` instead of `Vec` and added comments explaining the usage conditions.